src/main/java/Duke.java

```diff
@@ -14,25 +18,52 @@ public static void main(String[] args) {
                 + "\nHello! I'm Dook!"
                 + "\nWhat can I do for you?"
                 + "\n____________________________________________________________");
-        String[] userInputList = new String[100];
+        //this setup assumes the user is being cooperative
+        String[] taskList = new String[100];
+        Boolean[] markedTaskList = new Boolean[100];
         int userInputIndex = 0;
         Scanner scannerInput = new Scanner(System.in);
         boolean loopInput = true;
+        int markIndex;
         while (loopInput) {
             String userInput = scannerInput.nextLine();
-            switch (userInput) {
+            //split userinput by regex
+            Pattern commandPattern = Pattern.compile(COMMAND_FORMAT);
+            Matcher commandMatcher = commandPattern.matcher(userInput);
+            commandMatcher.find();
+            String commandInput = commandMatcher.group(1);
+            switch (commandInput) {
             case "bye":
                 loopInput = false;
                 break;
             case "list":
                 System.out.println("____________________________________________________________\n");
+                System.out.println("Here are the tasks in your list");
                 for (int i = 0; i<userInputIndex; i++) {
-                    System.out.println(String.valueOf(i+1)+". "+userInputList[i]);
+                    String printString = String.format("%d. [%c] %s", i+1, markedTaskList[i]?'X':' ', taskList[i]);
+                    System.out.println(printString);
```

Comment:
Maybe it is better to extract a method for each case?

Reply:
Thanks for the suggestion! I have refactored my code accordingly.